antarest/storage/service.py

```diff
@@ -182,22 +192,32 @@ def _save_metadata(
             metadata.groups = [group]
         self.repository.save(metadata)
 
-    def _get_metadata_according_permission(
-        self, user: Optional[User], uuid: str
-    ) -> Metadata:
-        if not user:
-            raise UserHasNotPermissionError()
+    def _get_metadata(self, uuid: str) -> Metadata:
 
         md = self.repository.get(uuid)
         if not md:
+            sanitized = StorageServiceUtils.sanitize(uuid)
             logger.warning(
                 f"Study %s not found in metadata db",
-                StorageServiceUtils.sanitize(uuid),
+                sanitized,
             )
+            raise StudyNotFoundError(uuid)
+        return md
+
+    def _assert_permission(
+        self,
+        user: Optional[User],
+        md: Optional[Metadata],
+        raising: bool = True,
+    ) -> bool:
+        if not user:
             raise UserHasNotPermissionError()
 
+        if not md:
+            raise ValueError("Metadata is None")
```

Comment:
This should be a StudyNotFoundError

Reply:
`StudyNotFoundError` is raising in `_get_metadata` because we search study in database. In this case, we don't not where come metadata. So if it none, we just said is none, we don't know why it is none....

Reply:
i had missed that, ok

Reply:
i had missed that, ok